tests/extra_source_and_tests/extra_src/shared/pressure_boundary/bidirectional_buffer.cpp

```diff
@@ -0,0 +1,37 @@
+#include "bidirectional_buffer.h"
+
+namespace SPH
+{
+//=====================================================================================================//
+namespace fluid_dynamics
+{
+//=================================================================================================//
+template <>
+void BidirectionalBuffer::Injection<execution::ParallelPolicy>::update(size_t index_i, Real dt)
+{
+    size_t sorted_index_i =buffer_particle_list_[index_i];
+        if (aligned_box_.checkUpperBound(axis_, pos_n_[sorted_index_i]) && buffer_particle_indicator_[sorted_index_i] == 1)
+        {
+            mutex_switch_to_real_.lock();
+            if (particles_->total_real_particles_ >= particles_->real_particles_bound_)
+            {
+                std::cout << "EmitterInflowBoundaryCondition::ConstraintAParticle: \n"
+                          << "Not enough body buffer particles! Exit the code."
+                          << "\n";
+                exit(0);
+            }
```

Comment:
For each particle iteration,  the "if" check is conducted, if possible to move the 'if check" out of the particle iteration loop to optimize the efficiency?

Reply:
I copy this if loop from the original open boundary code,  which is necessary and important